src/components/carousel/index.tsx

```diff
@@ -41,41 +42,42 @@ class Carousel extends Component<CarouselProps, CarouselState> {
     super(props);
 
     const defaultPageWidth = props.loop || !props.pageWidth ? Constants.screenWidth : props.pageWidth;
+    const pageHeight = props.pageHeight ?? Constants.screenHeight;
 
     this.state = {
       containerWidth: undefined,
       // @ts-ignore (defaultProps)
       currentPage: this.shouldUsePageWidth() ? this.getCalcIndex(props.initialPage) : props.initialPage,
       currentStandingPage: props.initialPage,
       pageWidth: defaultPageWidth,
-      initialOffset: {
-        x: presenter.calcOffset(props, {
-          // @ts-ignore (defaultProps)
-          currentPage: props.initialPage,
-          pageWidth: defaultPageWidth
-        }),
-        y: 0
-      },
+      pageHeight,
+      initialOffset: presenter.calcOffset(props, {
+        // @ts-ignore (defaultProps)
+        currentPage: props.initialPage,
+        pageWidth: defaultPageWidth,
+        pageHeight
+      }),
       prevProps: props
     };
   }
 
   static getDerivedStateFromProps(nextProps: CarouselProps, prevState: CarouselState) {
     const {currentPage, prevProps} = prevState;
-    const {pageWidth} = prevProps;
-    const {pageWidth: nextPageWidth} = nextProps;
+    const {pageWidth, pageHeight} = prevProps;
+    const {pageWidth: nextPageWidth, pageHeight: nextPageHeight} = nextProps;
 
-    if (!_.isUndefined(nextPageWidth) && pageWidth !== nextPageWidth) {
+    if ((!_.isUndefined(nextPageWidth) && pageWidth !== nextPageWidth)
+      && (!_.isUndefined(nextPageHeight) && pageHeight !== nextPageHeight)) {
```

Comment:
If I'm not wrong, these two conditions now depend on each other. 
If the user changes the `pageWidth` it should update the state regardless of if they change the `pageHeight`. 
They probably will only change one or the other..

Reply:
Changed it to `||`, hope that was your intention.

Reply:
Yes (: